Terminal.Gui/Core.cs

```diff
@@ -423,7 +423,8 @@ internal void SetNeedsLayout ()
 		/// </summary>
 		/// <param name="region">The region that must be flagged for repaint.</param>
 		public void SetNeedsDisplay (Rect region)
-		{
+		{			
+			SetNeedsLayout();
 			if (NeedDisplay.IsEmpty)
```

Comment:
This is not a good idea, as this will cause a very expensive computation on every request to redisplay a region.   The culprit for this problem is likely elsewhere.

SetNeedsLayout is intended to reduce the amount of work done, this does the opposite.

Conceptually this is not right either.

Reply:
This is the only solution I could find. Else will need to trigger `TerminalResized` as suggested by @IKoshelev. But I don't like that either as it is not the correct event. Can you please help me here? I am kind of stuck because of this. TIA.

Reply:
I added that because, as mentioned in the commit message, in `LayoutSubViews` function call `if layoutNeeded` is false which we set at the end of the same function call. For redrawing/repainting we need it to be true.